### PR TITLE
Shell: Propagate more Errors in AST.cpp

### PR DESCRIPTION
**Pull request by Hendiadyoin1**
Most functions there already had error handling capabilities, so let's just use TRY everywhere

### CODE REVIEW

File: Userland/Shell/AST.cpp [hunk 1]
```diff
@@ -319,17 +325,15 @@ ErrorOr<Vector<Command>> Node::to_lazy_evaluated_commands(RefPtr<Shell> shell)
 
 ErrorOr<void> Node::dump(int level) const
 {
-    print_indented(level,
+    return TRY(print_indented(level,
```
Comment (GMTA) [inline Userland/Shell/AST.cpp:328]: You can leave out this `TRY`

File: Userland/Shell/AST.cpp [hunk 2]
```diff
@@ -107,10 +107,16 @@ ErrorOr<void> AK::Formatter<Shell::AST::Command>::format(FormatBuilder& builder,
 namespace Shell::AST {
 
 template<typename... Args>
-static inline void print_indented(int indent, CheckedFormatString<Args...> format, Args&&... args)
+static inline ErrorOr<void> print_indented(int indent, CheckedFormatString<Args...> format, Args&&... args)
```
Comment (alimpfard) [inline Userland/Shell/AST.cpp:110]: I'm honestly not a fan of making debug paths TRY() things, but sure, whatever.

File: Userland/Shell/AST.cpp [hunk 3]
```diff
@@ -2295,39 +2298,36 @@ ErrorOr<RefPtr<Value>> MatchExpr::run(RefPtr<Shell> shell)
...
-                    spans.append(list[i].substring_from_byte_offset(span.start, span.length).release_value_but_fixme_should_propagate_errors());
+                    TRY(spans.try_append(TRY(list[i].substring_from_byte_offset(span.start, span.length))));
             }
 
             return true;
         }
     };
 
-    auto resolve_pattern = [&](auto& option) -> decltype(auto) {
+    auto resolve_pattern = [&](auto& option) -> Conditional<IsSame<RemoveCVReference<decltype(option)>, Regex<ECMA262>>, ErrorOr<Regex<ECMA262>>, ErrorOr<Vector<String>>> {
```
Comment (alimpfard) [inline Userland/Shell/AST.cpp:2308]: Yuck but okay.

File: Userland/Shell/AST.cpp [hunk 4]
```diff
@@ -2553,10 +2553,11 @@ ErrorOr<RefPtr<Value>> Pipe::run(RefPtr<Shell> shell)
...
     Vector<Command> commands;
-    commands.extend(left);
-    commands.append(last_in_left);
-    commands.append(first_in_right);
-    commands.extend(right);
+    TRY(commands.try_ensure_capacity(left.size() + 2 + right.size()));
+    MUST(commands.try_extend(left));
+    MUST(commands.try_append(last_in_left));
+    MUST(commands.try_append(first_in_right));
+    MUST(commands.try_extend(right));
```
Comment (alimpfard) [inline Userland/Shell/AST.cpp:2560]: `MUST(try_*())` sounds silly, I'm not sure what the consensus on this is, but I'd rather use the non-try functions here.

File: Userland/Shell/AST.cpp [hunk 5]
```diff
@@ -3748,11 +3750,13 @@ ErrorOr<String> StringValue::resolve_as_string(RefPtr<Shell> shell)
 ErrorOr<Vector<String>> StringValue::resolve_as_list(RefPtr<Shell> shell)
 {
     if (is_list()) {
+        // FIXME: Catch the allocation failure from StringView::split_view, by using
+        //        String::split(String const&,...)
```
Comment (alimpfard) [inline Userland/Shell/AST.cpp:3754]: That function is not a static function, why are you naming it like one?
also the fixme sounds superfluous, what's stopping this from happening?

**Review by alimpfard — COMMENTED** [review]
Thanks for working on this!

File: Userland/Shell/AST.cpp (hunk 2 above)
Comment (Hendiadyoin1) [reply Userland/Shell/AST.cpp:110]: Well all callers are TRY, but wouldnt actually throw, just call print_indented

File: Userland/Shell/AST.cpp (hunk 3 above)
Comment (Hendiadyoin1) [reply Userland/Shell/AST.cpp:2308]: I wasn't the person adding two different return types to a lambda....
I can clean it up to `ErrorOr<Conditional...>` if that is better

File: Userland/Shell/AST.cpp (hunk 4 above)
Comment (Hendiadyoin1) [reply Userland/Shell/AST.cpp:2560]: OK

File: Userland/Shell/AST.cpp (hunk 5 above)
Comment (Hendiadyoin1) [reply Userland/Shell/AST.cpp:3754]: The API is not there yet, and I could not be bothered to do a full rebuild, with changed String headers

Comment (Hendiadyoin1) [reply Userland/Shell/AST.cpp:3754]: Ideally we would use `for_each_split_view`, but then its either multiple allocations from append, or virtually two splits (which actually shouldn't be that expensive)
You can make a call and I'll change it accordingly

Comment (alimpfard) [reply Userland/Shell/AST.cpp:3754]: :thinking: 
sure is, `String.h:148`
```c++
    ErrorOr<Vector<String>> split(u32 separator, SplitBehavior = SplitBehavior::Nothing) const;
```

Comment (alimpfard) [reply Userland/Shell/AST.cpp:3754]: oh we're splitting by a string, got it

Comment (Hendiadyoin1) [reply Userland/Shell/AST.cpp]: I now changed it to MaxDue's

**Pull request closed without merging**